core/src/main/java/tachyon/conf/TachyonConf.java

```diff
@@ -142,7 +142,8 @@ protected void loadDefault(boolean includeSystemProperties) {
     Properties defaultProps = new Properties();
 
     // Override runtime default
-    defaultProps.setProperty(Constants.MASTER_HOSTNAME, NetworkUtils.getLocalHostName());
+    defaultProps.setProperty(Constants.MASTER_HOSTNAME, 
+        NetworkUtils.getLocalHostName(100));
```

Comment:
No, this is really to avoid having a long lookup every time a new `TachyonConf` instance is created.  If a host that resolves to the local machine is actually reachable we should be able to reach it pretty fast.

I will up this a bit